### DISCUSSION TITLE
[Major Update] Fooocus 2.0.0

### DISCUSSION THREAD

**Discussion by lllyasviel** [lllyasviel/Fooocus#347] [Announcements]
Fooocus has recently completely rewritten the text processing engine, and the big changes pushed the version directly to 2.0.0. 

1.	The new text feature processing engine begin to support multiple styles. Using multiple style often improves image quality in our tests. 
2.	In 100 tests (100 prompts written by ChatGPT), V2 default results outperform V1 default results in 87 cases, evaluated by two humans.
3.	In 100 tests (100 prompts written by ChatGPT), V2 prompt understanding outperform V1 prompt understanding in 81 cases, evaluated by two humans, in both default setting and single/multiple style mode.
4.	Because the above numbers are above 80%, we view this version as a major update and directly jump to 2.0.0.

Tips:
1.	“Prompt Expansion and Raw Mode” is renamed to “Fooocus V2” in Style. 
2.	(In most cases) you can reproduce previous V1 results (or V1.0.4 raw results) by turning off “Fooocus V2” in Style.
3.	`cinematic-default` is renamed to `Default (Slightly Cinematic)`. The content of this style does not change.
4.	All style names are standardized with space camel case for better UI.

**Comment by joanorsky** [comment]
This update has however made the T4 on collab to crash due to more CUDA memory requirements..

**Reply by lllyasviel** [reply]
i just tested and it seems still works

**Reply by joanorsky** [reply]
Humm... well i am using other models as well but .. on the T4 it crashes on my setup (stating memory allocation issues). On the other hand.. with a 3090 it works very well.. (maybe because i was using 24GB of VRAM)

**Reply by joanorsky** [reply]
I was just testing it .. and without a LoRA it seems to pass ok.. but once you put one in... the T4's VRAM becomes insufficient and it crashes..
![image](https://github.com/lllyasviel/Fooocus/assets/3766093/63f4f5f9-8bc3-4032-a4a3-6000166c602c)

**Reply by kubilaykilinc** [reply]
Yes, for me too, i am getting vram error. (win11, 16 GB RAM, 12 GB Vram Nvidia, 1 TB SSD)

**Comment by lee88688** [comment]
is current version supported cpu generate?

**Reply by WangzLoong** [reply]
不支持，而且目前只支持N卡

**Reply by tibelf** [reply]
P40 is ok

**Reply by markawonge** [reply]
AMD 7900XT通过ROCm满血跑的路过

**Comment by verifiedsyckozis** [comment]
Keeps stopping with the cuda issue.  Worked before the update.  Using a 3060 TI 8GB.  Seems to be fine before using a lora.  Just another person.

**Reply by lllyasviel** [reply]
i am looking into this

**Reply by lllyasviel** [reply]
I pushed a seemly fix see if works https://github.com/lllyasviel/Fooocus/pull/357

**Reply by joanorsky** [reply]
... going to test it in a few minutes as well.. already downloading a few LoRAs and models..

**Reply by joanorsky** [reply]
first run went ok with a T4 .. with a custom model and custom LoRA..
![image](https://github.com/lllyasviel/Fooocus/assets/3766093/8543e113-444e-4f27-bed0-9587a0948430)

**Reply by joanorsky** [reply]
Replacing LoRAs and models seems to be working as well.. at least on a T4...
![image](https://github.com/lllyasviel/Fooocus/assets/3766093/4368e297-ad3f-44f2-92b2-92d2678ad452)
... seems to be a lot more stable...

**Reply by joanorsky** [reply]
there seems to be some sort of memory leak though.. as the VRAM keeps increasing over time.
![image](https://github.com/lllyasviel/Fooocus/assets/3766093/cb2b4eb1-aabf-457d-8e05-817bd5817c03)

I have also noticed that there is a residual memory on the prompts.. because if you change the prompt, the next gen comes with the old prompt (i have not tested it much.. but i did noticed this several times)

**Reply by lllyasviel** [reply]
we do not know if code is correct or colab just influencing all SD related scripts (or, even all gradio related scripts)

**Reply by joanorsky** [reply]
![image](https://github.com/lllyasviel/Fooocus/assets/3766093/d47a2819-c2cf-4abe-85d0-a4a830dfc6e7)
yeah.. probably. I have not run it yet on a isolated machine to test it out... but there is some sort of memory leak on collab at least as the VRAM keeps increasing...

**Reply by lllyasviel** [reply]
2.0.17 is significantly improved on colab vram

**Reply by kubilaykilinc** [reply]
Thank you so much for your valuable effort.

**Comment by dxf123456** [comment]
I have a new problem, can anyone help me?

My computer configuration is as follows
![capture](https://github.com/lllyasviel/Fooocus/assets/65050387/2b15a97a-6232-4336-be15-4020aa8f19ec)
I can't work normally. What should I do to solve the problem?
![335522](https://github.com/lllyasviel/Fooocus/assets/65050387/ac60f5e0-0fad-4e4f-a26f-d31d44fe9b0d)
